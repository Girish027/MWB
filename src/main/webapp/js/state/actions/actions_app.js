
import _ from 'lodash';
import getUrl, { pathKey } from 'utils/apiUrls';
import { fetchUtil } from 'utils/fetchUtils';
import goToRoute from 'utils/routeHelpers';
import Constants from 'constants/Constants';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { logAmplitudeEvent } from 'utils/amplitudeUtils';
import axios from 'axios';
import api from 'utils/api';
import { setDefaultsForAPI } from 'utils/api';
import * as types from './types';

// Note to developers:
// Both Warning and Good Messages are shown for the period of notificationInterval.
// Bad, Persistant - till user closes the notification. This should only be used sparingly.
export const callLogIngestAPI = (message, logLevel) => {
  const promise = new Promise((resolve, reject) => {
    const url = getUrl(pathKey.logIngest, { logLevel });
    return new Promise((resolve, reject) => {
      api.post({
        url,
        headers: {
          'Content-Type': 'text/plain',
        },
        data: message,
      });
    });
  });
};

export const displayBadRequestMessage = (message) => (dispatch) => {
  dispatch({
    type: types.BAD_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_ERROR);
};

export const displayWarningRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.WARNING_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_WARNING);

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayGoodRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.GOOD_REQUEST,
    value: message,
  });

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayPersistantMessage = message => ({
  type: types.PERSISTENT_REQUEST,
  value: message,
});

export const stopShowingServerMessage = () => ({
  type: types.APP_STOP_SHOWING_SERVER_MESSAGE,
});

export const updateLoginSuccess = csrfToken => ({
  type: types.LOGIN_SUCCESS,
  value: csrfToken,
});

export const updateLoginInfo = (userId, userName, userDetails) => ({
  type: types.LOGIN_INFO,
  value: {
    userId,
    userName,
    userDetails,
  },
});

export const logoutFromOktaSuccess = () => ({
  type: types.LOGOUT_OKTA,
});

export const logoutFromMWBSuccess = () => ({
  type: types.LOGOUT_MWB,
});

export const contextMenuChange = contextMenuState => ({
  type: types.CONTEXT_MENU_CHANGE,
  contextMenuState,
});

export const modalDialogChange = modalDialogState => ({
  type: types.MODAL_DIALOG_CHANGE,
  modalDialogState,
});

const actionFileRemoveRequest = ({ fileId }) => ({
  type: types.FILE_REMOVE_REQUEST,
  fileId,
});

const actionFileRemoveError = ({ fileId, error }) => ({
  type: types.FILE_REMOVE_ERROR,
  fileId,
  error,
});

const actionFileRemoveSuccess = ({ fileId }) => ({
  type: types.FILE_REMOVE_SUCCESS,
  fileId,
});

export const updateUserGroups = data => ({
  type: types.UPDATE_USER_GROUPS,
  data,
});

const onFetchFailureRemoveFile = (error, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveError({ fileId, error }));
};

const onFetchSuccessRemoveFile = (data, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveSuccess({ fileId }));
};

export const removeFile = ({ fileId }) => (dispatch, getState) => {
  dispatch(actionFileRemoveRequest({ fileId }));

  const state = getState();
  const fetchUrl = getUrl(pathKey.files, { fileId, clientId: state.header.client.id });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'delete',
    onFetchSuccess: onFetchSuccessRemoveFile,
    onFetchFailure: onFetchFailureRemoveFile,
    dispatch,
    getState,
    params: {
      fileId,
    },
  });
};

const onFetchSuccessGetUserGroups = data => (dispatch) => {
  dispatch(updateUserGroups(data));
};

export const getUserGroups = ({ csrfToken }) => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.userGroups, { csrfToken });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetUserGroups,
    dispatch,
    getState,
  });
};

export const onFetchSuccessGetCSRFToken = data => (dispatch, getState) => {
  const csrfToken = data.value;
  setDefaultsForAPI(csrfToken);
  dispatch(updateLoginSuccess(csrfToken));
  dispatch(getUserGroups({ csrfToken }));
};

export const getCSRFToken = userId => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.csrfToken);

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetCSRFToken,
    dispatch,
    getState,
    params: {
      userId,
    },
  });
};

const logoutFromMWB = () => (dispatch, getState) => {
  dispatch(logoutFromOktaSuccess());
  dispatch(logoutFromMWBSuccess());
};

export const logoutFromOkta = () => (dispatch, getState) => {
  const state = getState();
  const { oAuthLogoutURL } = state.app;
  if (oAuthLogoutURL) {
    fetch(oAuthLogoutURL, {
      method: 'DELETE',
      credentials: 'include',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    })
      .then((response) => {
        window.location.href = Constants.LOGOUT;
        switch (response.status) {
        case 204:
          logoutFromMWB();
          logAmplitudeEvent(AmplitudeConstants.LOGOUT_EVENT, state);
          break;
        case 404:
        default:
          // TODO send log for these scenarios
          break;
        }
      })
      .catch((error) => {
        // TODO send log for these scenarios
      });
  }
};

export const changeRoute = (routeName, data, history) => (dispatch, getState) => {
  const state = getState();
  const routeData = Object.assign({}, {
    client: !_.isNil(state.header.client) ? state.header.client : {
      id: 0,
      name: 'Loading...',
    },
    projectId: state.selectedProjectId,
  }, data);
  goToRoute(routeName, routeData, history);
};
