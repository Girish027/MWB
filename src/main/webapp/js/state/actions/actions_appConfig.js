import 'regenerator-runtime/runtime';

import getUrl, { pathKey } from 'utils/apiUrls';

import { fetchUtil } from 'utils/fetchUtils';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { initializeAnalytics, logAmplitudeEvent } from 'utils/amplitudeUtils';
import api from 'utils/api';
import * as types from './types';

const appConfigSuccess = data => ({
  type: types.APP_CONFIG_SUCCESS,
  data,
});

const appConfigFailure = data => ({
  type: types.APP_CONFIG_FAILURE,
  data,
});

const appConfigRequest = () => ({
  type: types.APP_CONFIG_REQUEST,
});

export const onAppConfigSuccess = data => (dispatch, getState) => {
  dispatch(appConfigSuccess(data));
  const state = getState();
  initializeAnalytics(state);
  logAmplitudeEvent(AmplitudeConstants.LOGIN_EVENT, state);
};

export const onAppConfigFailure = error => (dispatch) => {
  dispatch(appConfigFailure(error));
};

export const fetchAppConfig = () => (dispatch, getState) => {
  dispatch(appConfigRequest());
  const fetchUrl = getUrl(pathKey.appConfig);

  api.get({
    url: fetchUrl,
    onApiSuccess: onAppConfigSuccess,
    onApiError: onAppConfigFailure,
    dispatch,
    getState,
  });
};
