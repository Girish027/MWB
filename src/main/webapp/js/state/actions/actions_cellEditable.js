import * as types from './types';

export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_CREATE,
  stateKey,
});

export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_REMOVE,
  stateKey,
});

export const setActiveCell = ({ stateKey, activeRowIndex, activeColumnIndex }) => ({
  type: types.CELL_EDITABLE_SET_ACTIVE_CELL,
  stateKey,
  activeRowIndex,
  activeColumnIndex,
});

export const updateValue = ({ stateKey, value }) => ({
  type: types.CELL_EDITABLE_UPDATE_VALUE,
  stateKey,
  value,
});
