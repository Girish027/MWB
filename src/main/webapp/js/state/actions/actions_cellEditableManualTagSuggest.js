import store from 'state/configureStore';
import errorMessageUtil from 'utils/ErrorMessageUtil';
import getUrl, { pathKey } from 'utils/apiUrls';
import * as types from './types';

let controller;
export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_CREATE,
  stateKey,
});
export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_REMOVE,
  stateKey,
});

const getSuggestedTagsRequest = ({ stateKey, projectId, intent }) => ({
  type: types.GET_SUGGESTED_TAGS_REQUEST,
  stateKey,
  projectId,
  intent,
});

const getSuggestedTagsError = ({
  stateKey, projectId, intent, error,
}) => ({
  type: types.GET_SUGGESTED_TAGS_ERROR,
  stateKey,
  projectId,
  intent,
  error,
});

const getSuggestedTagsSuccess = ({
  stateKey, projectId, intent, suggestedTags,
}) => ({
  type: types.GET_SUGGESTED_TAGS_SUCCESS,
  stateKey,
  projectId,
  intent,
  suggestedTags,
});

export const getSuggestedTags = ({ stateKey, projectId, intent }) => (dispatch, getState) => new Promise((resolve, reject) => {
  if (controller) {
    controller.abort();
  }
  controller = new AbortController();
  const signal = controller.signal;
  dispatch(getSuggestedTagsRequest({ stateKey, projectId, intent }));
  const state = getState();
  const { csrfToken } = state.app;
  const clientId = state.header.client.id;

  const fetchUrl = getUrl(pathKey.suggestedTags, { projectId, intent, clientId });

  fetch(fetchUrl, {
    signal,
    credentials: 'same-origin',
    headers: {
      'X-CSRF-TOKEN': csrfToken,
    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((json) => {
      const suggestedTags = json && json.length ? json : [];
      const intentIndex = suggestedTags.indexOf(intent);
      if (intentIndex >= 0) {
        suggestedTags.slice(intentIndex, 1);
      }
      dispatch(getSuggestedTagsSuccess({
        stateKey, projectId, intent, suggestedTags,
      }));
      resolve([...suggestedTags]);
    })
    .catch((error) => {
      if (error.name === 'AbortError') { return; }
      dispatch(getSuggestedTagsError({
        stateKey, projectId, intent, error,
      }));
      reject(error);
    });
});

export const cursorDown = ({ stateKey }) => ({
  type: types.CELL_SUGGEST_CURSOR_DOWN,
  stateKey,
});

export const cursorUp = ({ stateKey }) => ({
  type: types.CELL_SUGGEST_CURSOR_UP,
  stateKey,
});
