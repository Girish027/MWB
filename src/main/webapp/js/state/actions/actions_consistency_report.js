import errorMessageUtil from 'utils/ErrorMessageUtil';
import store from 'state/configureStore';

import getUrl, { pathKey } from 'utils/apiUrls';
import * as appActions from './actions_app';
import * as types from './types';

export const reset = () => ({
  type: types.CONSISTENCY_REPORT_RESET,
});

export const setProjectId = projectId => ({
  type: types.CONSISTENCY_REPORT_SET_PROJECT_ID,
  projectId,
});

export const setProject = ({ projectId, project }) => ({
  type: types.CONSISTENCY_REPORT_SET_PROJECT,
  projectId,
  project,
});

export const setDatasets = ({ projectId, datasets }) => ({
  type: types.CONSISTENCY_REPORT_SET_DATASETS,
  projectId,
  datasets,
});


export const setFilter = ({ datasets, onlyConflicts }) => ({
  type: types.CONSISTENCY_REPORT_SET_FILTER,
  datasets,
  onlyConflicts,
});

const requestSearchSuccess = ({ results, startIndex }) => ({
  type: types.CONSISTENCY_REPORT_REQUEST_SEARCH_SUCCESS,
  results,
  startIndex,
});

const requestSearchRequest = ({
  promise, projectId, query, filter, sort, startIndex, limit,
}) => ({
  type: types.CONSISTENCY_REPORT_REQUEST_SEARCH_REQUEST,
  promise,
  projectId,
  query,
  filter,
  sort,
  startIndex,
  limit,
});

const requestSearchError = ({ error }) => ({
  type: types.CONSISTENCY_REPORT_REQUEST_SEARCH_ERROR,
  error,
});

export const setPagerSettings = ({ startIndex, limit, showControls }) => ({
  type: types.CONSISTENCY_REPORT_SET_PAGER_SETTINGS,
  startIndex,
  limit,
  showControls,
});

export const requestSearch = ({
  projectId, query, filter, sort, startIndex, limit, delay,
}, isNested) => (dispatch, getState) => {
  const promise = new Promise((resolve, reject) => {
    const state = getState();
    const { userId, csrfToken } = state.app;
    const clientId = state.header.client.id;

    const fetchUrl = getUrl(pathKey.consistency, {
      projectId,
      startIndex,
      limit,
      sortProperty: sort.property,
      sortDirection: sort.direction,
      clientId,
    });

    const filterBodyObj = {};
    filterBodyObj.onlyTagged = true;
    filterBodyObj.onlyConflicts = filter.onlyConflicts;
    if (filter.datasets && filter.datasets.length) {
      filterBodyObj.datasetIds = filter.datasets;
    }

    setTimeout(() => {
      fetch(fetchUrl, {
        credentials: 'same-origin',
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'X-CSRF-TOKEN': csrfToken,
        },
        body: JSON.stringify({
          configId: '0',
          filter: filterBodyObj,
        }),
      })
        .then(errorMessageUtil.handleErrors)
        .then(response => response.json())
        .then((results) => {
          const searchResults = results.transcriptionList || [];

          searchResults.forEach((res, index) => {
            res._rowIndex = index;
          });

          if (!searchResults.length && startIndex > 0) {
            const total = results.total;
            const maxPage = Math.max(Math.floor(total / limit) + (total % limit ? 1 : 0), 1);
            const newStartIndex = (maxPage - 1) * limit;
            dispatch(requestSearch({
              projectId,
              query,
              filter,
              sort,
              startIndex: newStartIndex,
              limit,
            }, true))
              .then((results) => {
                dispatch(requestSearchSuccess({ results, startIndex: newStartIndex }));
                resolve(results);
              })
              .catch((error) => {
                reject(error);
              });
          } else {
            if (!isNested) {
              dispatch(requestSearchSuccess({ results, startIndex }));
            }
            resolve(results);
          }
        })
        .catch((error) => {
          dispatch(requestSearchError({ error }));
          errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage);
          reject(error);
        });
    }, delay || 0);
  });

  if (!isNested) {
    dispatch(requestSearchRequest({
      promise, projectId, query, filter, sort, startIndex, limit,
    }));
  }

  return promise;
};

export const refresh = ({ delay }) => ({
  type: types.CONSISTENCY_REPORT_REFRESH,
  delay,
});
