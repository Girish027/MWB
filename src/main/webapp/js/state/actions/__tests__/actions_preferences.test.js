jest.mock('utils/api');

import api from 'utils/api';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import getUrl, { pathKey } from 'utils/apiUrls';
import * as actionsPreferences from 'state/actions/actions_preferences';

const middlewares = [thunk];
const reduxMockStore = configureMockStore(middlewares);


describe('actions_preferences', () => {
  let store;

  const clientLevelPreference = {
    id: 3,
    client_id: 151,
    level: 'client',
    type: 'use_large',
    attribute: '151',
    value: 2,
  };
  const allPreferences = [{
    id: 3,
    client_id: 151,
    level: 'client',
    type: 'use_large',
    attribute: '151',
    value: 1,
  }, {
    id: 4,
    client_id: 151,
    level: 'model',
    type: 'n-gram',
    attribute: '151',
    value: 1,
  }];
  const vectorizer = {
    N_GRAM: 1,
    USE: 2,
  };
  const currentType = {
    id: 2,
    type: 'use_large',
    version: 1.2,
    isLatest: 1,
  };

  beforeAll(() => {
    store = reduxMockStore({
      app: {
        userId: 'testUser',
      },
      header: {
        client: {
          id: '123',
        },
      },
      preferences: {
        technology: '',
        clientId: '123',
        isClientAvailable: true,
        clientLevelPreference,
        allPreferences,
        vectorizer,
        currentType,
      },
      projectListSidebar: {
        projectById: {
          111: {
            name: 'TestName',
          },
        },
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
    store.clearActions();
  });

  test('should call getTechnology', () => {
    store.dispatch(actionsPreferences.getTechnology({ clientId: '123' }));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call onGetTechnologySuccess', () => {
    store.dispatch(actionsPreferences.onGetTechnologySuccess(undefined, allPreferences));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call onGetTechnologyFailure', () => {
    store.dispatch(actionsPreferences.onGetTechnologyFailure());
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call technologySelected', () => {
    store.dispatch(actionsPreferences.technologySelected({}));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call addOrUpdateTechnology', () => {
    store.dispatch(actionsPreferences.addOrUpdateTechnology({ technologyValue: 'n-gram', updateExistingModels: false }));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call onUpdateModelTechnologySuccess', () => {
    const apiResponse = [{
      id: 4,
      client_id: 151,
      level: 'model',
      type: 'use_large',
      attribute: '151',
      value: 1,
    }];
    const id = apiResponse[0].id;
    store.dispatch(actionsPreferences.onUpdateModelTechnologySuccess(apiResponse, id, allPreferences, vectorizer));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call onUpdateModelTechnologySuccess with empty allPreferences', () => {
    const apiResponse = [{
      id: 4,
      client_id: 151,
      level: 'model',
      type: 'use_large',
      attribute: '151',
      value: 1,
    }];
    const id = apiResponse[0].id;
    const allPreferences = [];
    store.dispatch(actionsPreferences.onUpdateModelTechnologySuccess(apiResponse, id, allPreferences, vectorizer));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call onUpdateModelTechnologyFailure', () => {
    store.dispatch(actionsPreferences.onUpdateModelTechnologyFailure(undefined, allPreferences, vectorizer));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call updateModelTechnology with use_large', () => {
    const modelsResponse = [{
      id: 3,
      client_id: 151,
      level: 'model',
      type: 'use_large',
      attribute: '151',
      value: 2,
    }];
    store.dispatch(actionsPreferences.updateModelTechnology({ technologyValue: 'use_large', modelsResponse }));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call updateModelTechnology with N-Gram', () => {
    const modelsResponse = [{
      id: 3,
      client_id: 151,
      level: 'model',
      type: 'n-gram',
      attribute: '151',
      value: 2,
    }];
    store.dispatch(actionsPreferences.updateModelTechnology({ technologyValue: 'n-gram', modelsResponse }));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call updateClientTechnology with use_large', () => {
    store.dispatch(actionsPreferences.updateClientTechnology({ technologyValue: 'use_large', id: '3' }));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call updateClientTechnology with n-gram', () => {
    store.dispatch(actionsPreferences.updateClientTechnology({ technologyValue: 'n-gram', id: '3' }));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call updateClientTechnology with empty id', () => {
    store.dispatch(actionsPreferences.updateClientTechnology({ technologyValue: 'use_large', id: '' }));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call addClientTechnology', () => {
    store.dispatch(actionsPreferences.addClientTechnology({ technologyValue: 'use_large' }));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call onAddOrUpdateClientTechnologySuccess', () => {
    const apiResponse = clientLevelPreference;
    store.dispatch(actionsPreferences.onAddOrUpdateClientTechnologySuccess(
      apiResponse, allPreferences, vectorizer,
    ));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call onAddOrUpdateClientTechnologySuccess with no data in db', () => {
    const allPreferences = [];
    const apiResponse = clientLevelPreference;
    store.dispatch(actionsPreferences.onAddOrUpdateClientTechnologySuccess(
      apiResponse, allPreferences, vectorizer,
    ));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call onAddOrUpdateClientTechnologyFailure', () => {
    store.dispatch(actionsPreferences.onAddOrUpdateClientTechnologyFailure(undefined, allPreferences, vectorizer));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call getVectorizer', () => {
    store.dispatch(actionsPreferences.getVectorizer());
    expect(store.getActions()).toMatchSnapshot();
  });


  test('should call onGetVectorizerSuccess', () => {
    const response = [{
      id: 1,
      type: 'n-gram',
    },
    {
      id: 2,
      type: 'use_large',
      isLatest: 1,
    },
    ];
    store.dispatch(actionsPreferences.onGetVectorizerSuccess(response));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call onGetVectorizerFailure', () => {
    store.dispatch(actionsPreferences.onGetVectorizerFailure());
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call getTechnologyByClientModel', () => {
    store.dispatch(actionsPreferences.getTechnologyByClientModel({ clientId: '123', projectId: '123' }));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call getTechnologyByClientModelResult', () => {
    const response = {
      id: 2,
      type: 'use_large',
      version: 1.2,
      isLatest: 1,
    };
    store.dispatch(actionsPreferences.getVectorizerByClientProjectResult(response));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call createModelPreferences', () => {
    store.dispatch(actionsPreferences.createModelPreferences({ technologyValue: 'use_large', modelsResponse: ['111'] }));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call createModelPreferencesSuccess', () => {
    const apiResponse = {
      id: 5,
      client_id: 151,
      level: 'model',
      type: 'use_large',
      attribute: '151',
      value: 2,
    };
    store.dispatch(actionsPreferences.onCreateModelPreferencesSuccess(apiResponse, allPreferences, vectorizer));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call createModelPreferencesFailure', () => {
    store.dispatch(actionsPreferences.onCreateModelPreferencesFailure(undefined, allPreferences, vectorizer));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call addOrUpdateTechnology for technology with use', () => {
    store.dispatch(actionsPreferences.addOrUpdateTechnology({ technologyValue: 'use_large', updateExistingModels: true }));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call addOrUpdateTechnology for technology with ngram', () => {
    store.dispatch(actionsPreferences.addOrUpdateTechnology({ technologyValue: 'n-gram', updateExistingModels: true }));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('should call createModelPreferences', () => {
    store.dispatch(actionsPreferences.createModelPreferences({ technologyValue: 'use_large', modelsResponse: ['111'] }));
    const testUrl = '/nltools/private/v1/preference?clientId=151&setDefault=true';
    expect(api.post).toHaveBeenCalledTimes(1);
    expect(api.post.mock.calls[0][0].testUrl).toMatchSnapshot();
  });
});
