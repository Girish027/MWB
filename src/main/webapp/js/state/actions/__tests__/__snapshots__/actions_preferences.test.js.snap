// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions_preferences should call addClientTechnology 1`] = `Array []`;

exports[`actions_preferences should call addOrUpdateTechnology 1`] = `Array []`;

exports[`actions_preferences should call addOrUpdateTechnology for technology with ngram 1`] = `Array []`;

exports[`actions_preferences should call addOrUpdateTechnology for technology with use 1`] = `Array []`;

exports[`actions_preferences should call createModelPreferences 1`] = `Array []`;

exports[`actions_preferences should call createModelPreferences 2`] = `undefined`;

exports[`actions_preferences should call createModelPreferencesFailure 1`] = `
Array [
  Object {
    "allPreferences": Array [],
    "clientLevelPreference": Object {},
    "clientType": undefined,
    "isClientAvailable": false,
    "type": "DEFAULT_TECHNOLOGY",
    "vectorizer": Object {},
  },
]
`;

exports[`actions_preferences should call createModelPreferencesSuccess 1`] = `
Array [
  Object {
    "allPreferences": Array [
      Object {
        "attribute": "151",
        "client_id": 151,
        "id": 5,
        "level": "model",
        "type": "use_large",
        "value": 2,
      },
    ],
    "clientLevelPreference": Object {},
    "clientType": undefined,
    "isClientAvailable": true,
    "type": "DEFAULT_TECHNOLOGY",
    "vectorizer": Object {},
  },
]
`;

exports[`actions_preferences should call getTechnology 1`] = `Array []`;

exports[`actions_preferences should call getTechnologyByClientModel 1`] = `Array []`;

exports[`actions_preferences should call getTechnologyByClientModelResult 1`] = `
Array [
  Object {
    "response": Object {
      "id": 2,
      "isLatest": 1,
      "type": "use_large",
      "version": 1.2,
    },
    "type": "RECEIVE_PROJECT_PREFERENCE",
  },
]
`;

exports[`actions_preferences should call getVectorizer 1`] = `Array []`;

exports[`actions_preferences should call onAddOrUpdateClientTechnologyFailure 1`] = `
Array [
  Object {
    "allPreferences": Array [],
    "clientLevelPreference": Object {},
    "clientType": undefined,
    "isClientAvailable": false,
    "type": "DEFAULT_TECHNOLOGY",
    "vectorizer": Object {},
  },
]
`;

exports[`actions_preferences should call onAddOrUpdateClientTechnologySuccess 1`] = `
Array [
  Object {
    "allPreferences": Array [
      Object {
        "attribute": "151",
        "client_id": 151,
        "id": 3,
        "level": "client",
        "type": "use_large",
        "value": 2,
      },
    ],
    "clientLevelPreference": Object {
      "attribute": "151",
      "client_id": 151,
      "id": 3,
      "level": "client",
      "type": "use_large",
      "value": 2,
    },
    "clientType": undefined,
    "isClientAvailable": true,
    "type": "DEFAULT_TECHNOLOGY",
    "vectorizer": Object {},
  },
  Object {
    "type": "GOOD_REQUEST",
    "value": "Model default is successfully updated to n-grams",
  },
]
`;

exports[`actions_preferences should call onAddOrUpdateClientTechnologySuccess with no data in db 1`] = `
Array [
  Object {
    "allPreferences": Array [
      Object {
        "attribute": "151",
        "client_id": 151,
        "id": 3,
        "level": "client",
        "type": "use_large",
        "value": 2,
      },
    ],
    "clientLevelPreference": Object {
      "attribute": "151",
      "client_id": 151,
      "id": 3,
      "level": "client",
      "type": "use_large",
      "value": 2,
    },
    "clientType": undefined,
    "isClientAvailable": true,
    "type": "DEFAULT_TECHNOLOGY",
    "vectorizer": Object {},
  },
  Object {
    "type": "GOOD_REQUEST",
    "value": "Model default is successfully updated to n-grams",
  },
]
`;

exports[`actions_preferences should call onGetTechnologyFailure 1`] = `
Array [
  Object {
    "allPreferences": Array [],
    "clientLevelPreference": Object {},
    "clientType": "n-gram",
    "isClientAvailable": false,
    "type": "DEFAULT_TECHNOLOGY",
    "vectorizer": undefined,
  },
]
`;

exports[`actions_preferences should call onGetTechnologySuccess 1`] = `
Array [
  Object {
    "allPreferences": Array [],
    "clientLevelPreference": Object {},
    "clientType": "n-gram",
    "isClientAvailable": false,
    "type": "DEFAULT_TECHNOLOGY",
    "vectorizer": undefined,
  },
]
`;

exports[`actions_preferences should call onGetVectorizerFailure 1`] = `
Array [
  Object {
    "allPreferences": Array [
      Object {
        "attribute": "151",
        "client_id": 151,
        "id": 3,
        "level": "client",
        "type": "use_large",
        "value": 1,
      },
      Object {
        "attribute": "151",
        "client_id": 151,
        "id": 4,
        "level": "model",
        "type": "n-gram",
        "value": 1,
      },
    ],
    "clientLevelPreference": Object {},
    "clientType": undefined,
    "isClientAvailable": false,
    "type": "DEFAULT_TECHNOLOGY",
    "vectorizer": Object {
      "N_GRAM": 1,
      "USE": 2,
    },
  },
]
`;

exports[`actions_preferences should call onGetVectorizerSuccess 1`] = `
Array [
  Object {
    "allPreferences": Array [
      Object {
        "attribute": "151",
        "client_id": 151,
        "id": 3,
        "level": "client",
        "type": "use_large",
        "value": 1,
      },
      Object {
        "attribute": "151",
        "client_id": 151,
        "id": 4,
        "level": "model",
        "type": "n-gram",
        "value": 1,
      },
    ],
    "clientLevelPreference": Object {
      "attribute": "151",
      "client_id": 151,
      "id": 3,
      "level": "client",
      "type": "use_large",
      "value": 2,
    },
    "clientType": "use_large",
    "isClientAvailable": true,
    "type": "DEFAULT_TECHNOLOGY",
    "vectorizer": Object {
      "N_GRAM": 1,
      "USE": 2,
    },
  },
]
`;

exports[`actions_preferences should call onUpdateModelTechnologyFailure 1`] = `
Array [
  Object {
    "allPreferences": Array [],
    "clientLevelPreference": Object {},
    "clientType": undefined,
    "isClientAvailable": false,
    "type": "DEFAULT_TECHNOLOGY",
    "vectorizer": Object {},
  },
]
`;

exports[`actions_preferences should call onUpdateModelTechnologySuccess 1`] = `
Array [
  Object {
    "allPreferences": Array [
      Array [
        Object {
          "attribute": "151",
          "client_id": 151,
          "id": 4,
          "level": "model",
          "type": "use_large",
          "value": 1,
        },
      ],
    ],
    "clientLevelPreference": Object {},
    "clientType": undefined,
    "isClientAvailable": true,
    "type": "DEFAULT_TECHNOLOGY",
    "vectorizer": Object {},
  },
]
`;

exports[`actions_preferences should call onUpdateModelTechnologySuccess with empty allPreferences 1`] = `
Array [
  Object {
    "allPreferences": Array [
      Array [
        Object {
          "attribute": "151",
          "client_id": 151,
          "id": 4,
          "level": "model",
          "type": "use_large",
          "value": 1,
        },
      ],
    ],
    "clientLevelPreference": Object {},
    "clientType": undefined,
    "isClientAvailable": true,
    "type": "DEFAULT_TECHNOLOGY",
    "vectorizer": Object {},
  },
]
`;

exports[`actions_preferences should call technologySelected 1`] = `
Array [
  Object {
    "allPreferences": Array [],
    "clientLevelPreference": Object {},
    "clientType": undefined,
    "isClientAvailable": false,
    "type": "DEFAULT_TECHNOLOGY",
    "vectorizer": undefined,
  },
]
`;

exports[`actions_preferences should call updateClientTechnology with empty id 1`] = `Array []`;

exports[`actions_preferences should call updateClientTechnology with n-gram 1`] = `Array []`;

exports[`actions_preferences should call updateClientTechnology with use_large 1`] = `Array []`;

exports[`actions_preferences should call updateModelTechnology with N-Gram 1`] = `Array []`;

exports[`actions_preferences should call updateModelTechnology with use_large 1`] = `Array []`;
